### DISCUSSION TITLE
Simple topic types [13814]

### DISCUSSION THREAD

**Discussion by NeilTruter** [eProsima/Fast-DDS#2411] [Q&A] [answered]
Good day,
Must all types used by topics be defined within the idl file?
If I want to create a topic that is only of type int8, must I define a type in the idl file and then register that type, or can I just register the int8 as a type and create the topic, or are there build-in registered types for stuff like int8, boolean etc?

**Comment by EduPonz** [comment, marked as answer]
Hi @NeilTruter ,

You could go around the IDL + Fast DDS-Gen process using [Dynamic Types](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dynamic_types/dynamic_types.html). With this API, you can define your types programmatically. for your use-case, you would need a [primitive types](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dynamic_types/supported_types.html#primitive-types). You can check several examples using this API:
* [Dynamic types example](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dynamic_types/examples.html)
* [Dynamic HelloWorld example](https://github.com/eProsima/Fast-DDS/tree/master/examples/C%2B%2B/DDS/DynamicHelloWorldExample)
* A totally unrelated [snippet](https://fast-dds.docs.eprosima.com/en/latest/fastdds/persistence/persistence.html#example) that uses this API.
